pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -104,9 +110,62 @@ func getPreFilterState(cycleState *framework.CycleState) (preFilterState, error)
 func (pl *NodePorts) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
 		// Due to immutable fields `spec.containers[*].ports`, pod update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
+	}
+}
+
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *NodePorts) isSchedulableAfterPodDeleted(pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	deletedPod, _, err := util.As[*v1.Pod](oldObj, nil)
+	if err != nil {
+		pl.logger.Error(err, "unexpected object in isSchedulableAfterPodDeleted", "oldObj", oldObj)
+		return framework.QueueAfterBackoff
+	}
+
+	// If the deleted pod is unscheduled, it doesn't make the target pod schedulable.
+	if deletedPod.Spec.NodeName == "" {
```

Comment:
Hi @sanposhiho, I raise a [pr](https://github.com/kubernetes/kubernetes/pull/119191) about the deletion event of pods. The `queue.AssignedPodDelete` literally means that an assigned pod was deleted. Howerver, If pod is hold by the cache.DeletedFinalStateUnknown, it maybe in unassigned state. 

FYI: https://github.com/carlory/kubernetes/blob/4ed12f217d12a9d4698ff087895c9c5aae03e09e/pkg/scheduler/eventhandlers.go#L327

```go
	// scheduled pod cache
	if handlerRegistration, err = informerFactory.Core().V1().Pods().Informer().AddEventHandler(
		cache.FilteringResourceEventHandler{
			FilterFunc: func(obj interface{}) bool {
				switch t := obj.(type) {
				case *v1.Pod:
					return assignedPod(t)
				case cache.DeletedFinalStateUnknown:
					if _, ok := t.Obj.(*v1.Pod); ok {
						// The carried object may be stale, so we don't use it to check if
						// it's assigned or not. Attempting to cleanup anyways.
						return true
					}
					utilruntime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Pod in %T", obj, sched))
					return false
				default:
					utilruntime.HandleError(fmt.Errorf("unable to handle object in %T: %T", sched, obj))
					return false
				}
			},
			Handler: cache.ResourceEventHandlerFuncs{
				AddFunc:    sched.addPodToCache,
				UpdateFunc: sched.updatePodInCache,
				DeleteFunc: sched.deletePodFromCache,
			},
		},
	); err != nil {
		return err
	}
	handlers = append(handlers, handlerRegistration)
```

So I think we should check the pod's state before we move unscheduable other pods to ativeQ or backOffQ. how about this? Can you help me review it.

Reply:
Should I check if the deleted pod is assigned? I'm not sure if an unassigned pod make an other pod schedulable.

Reply:
> Should I check if the deleted pod is assigned?

We don't need to check the deleted event if the deleted pod is not assigned to any Node. So, the current impl looks good on that point.